src/inspector_agent.cc

```diff
@@ -477,6 +488,40 @@ void AgentImpl::InstallInspectorOnProcess() {
   v8::Local<v8::Object> inspector = v8::Object::New(env->isolate());
   READONLY_PROPERTY(process, "inspector", inspector);
   env->SetMethod(inspector, "wrapConsoleCall", InspectorWrapConsoleCall);
+  if (options_.wait_for_connect()) {
+    env->SetMethod(inspector, "callAndPauseOnStart", CallAndPauseOnStart);
+  }
+}
+
+// static
+void AgentImpl::CallAndPauseOnStart(
+      const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  v8::HandleScope scope(env->isolate());
+  if (args.Length() < 2) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function and the second one should be "
+                           "a thisArg");
+
+  } else if (!args[0]->IsFunction()) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function");
+  }
+
+  std::vector<v8::Local<v8::Value>> call_args;
+  for (int i = 2; i < args.Length(); i++) {
+    call_args.push_back(args[i]);
+  }
+
+  env->inspector_agent()->SchedulePauseOnNextStatement("Break on start");
```

Comment:
This effectively breaks on the next call into JS land?  If this function were to return instead of calling the function, it would break at the next line in lib/module.js?